test/test_commands/test_response_parsing.py

```diff
@@ -0,0 +1,345 @@
+"""
+Unit tests for lean server response classes
+
+Test that responses are properly converted from JSON into
+Python classes and that new fields do not cause the parser
+to crash.
+
+When possible, tests should use ACTUAL LEAN OUTPUT under a
+range of scenarios to ensure that all cases are covered.
+(One way to generate output is to use the trio server with
+debug_bytes=True.)
+"""
+import json
+
+import lean_client.commands as cmds
+
+
+class TestAllMessagesResponse:
+    def test_no_messages(self):
+        response_json = '{"msgs":[],"response":"all_messages"}'
+        resp = cmds.Response.parse_response(response_json)
+
+        assert isinstance(resp, cmds.AllMessagesResponse)
```

Comment:
It's usually a bit easier to compare the whole object at once here instead of all 3 assertions -- i.e. 
```suggestion
        assert resp == cmds.AllMessagesResponse(msgs=[])
```

and below.